src/core/fft_helpers.cpp

```diff
@@ -66,6 +66,7 @@ int normalize(const float *abs_spectrum, float *norm_spectrum, unsigned int bin_
 	if (abs_spectrum == NULL || norm_spectrum == NULL) {return -1;}
 	if (bin_count == 0 || block_size == 0) {return -1;}
 
+	block_size /= 2;
```

Comment:
The input block has a given size, say 2048 samples. Then FFTW returns a buffer of 1024 (or 1025?) samples, because the rest is symmetric. The result is not normalized and is in fact multiplied by _N_, which is the number of either blocks or bins (I don't remember). In any case, the normalization of full scale sine wave previously produced peak value of -3 dB, which is not the desired value (it should be 0 dB to bring it in line of most other analyzers).

To get the normalized value every sample is divided by _N_. It just turns out I was using the wrong _N_ before.

Reply:
Oh, ofc, spectrum size is halve of block size, so you divide by 2. No, OK, I think it's clean then.